network/src/network.rs

```diff
@@ -1024,6 +1035,15 @@ impl NetworkService {
                     warn!("send shutdown message to p2p error: {:?}", err);
                 }
 
+                debug!("Dumping peer store data ...");
```

Comment:
Please implement the `Drop` trait for `DumpPeerStoreService` and invoke the `dump_peer_store` in the drop fn.

Reply:
Nope, it's bad to do this in Drop,  it's implicit and tricky and hard to handle error.

Reply:
Update, dump peer store when `DumpPeerStoreService` receives an exit signal.